code/modules/paperwork/paperbin.dm

```diff
@@ -120,50 +144,92 @@
 
 /obj/item/paper_bin/update_overlays()
 	. = ..()
+
+	total_paper = LAZYLEN(papers)
+
+	if(bin_pen)
+		pen_overlay = mutable_appearance(bin_pen.icon, bin_pen.icon_state)
+
+	if(!bin_overlay)
+		bin_overlay = mutable_appearance(icon, bin_overlay_string)
+
+	if(LAZYLEN(papers))
+		var/paper_number = 1
+		for(var/obj/item/paper/current_paper in papers)
```

Comment:
why not make this a `for(var/i in 1 to papers.len)` sort of deal?

Reply:
Did it, thanks